Vault entity fix: Add table with values that can be stored as secrets (#385)
Kong/developer.konghq.com: app/_gateway_entities/vault.md

Reviewer: One minor comment: now that there's a section for the supported values, the list at lines 50-55 is redundant and can be removed.

    supports_konnect: true
{% endfeature_table %}

## What can be stored as a secret?

You can store and reference the following fields as secrets in a Vault:

{% feature_table %}
item_title: Field type
columns:
  - title: {{site.konnect_short_name}} Config Store
    key: konnect
  - title: Env variable vault
    key: env
  - title: HashiCorp Vault
    key: hcv
  - title: Google Cloud
    key: gcp
  - title: AWS
    key: aws
  - title: Azure
    key: azure

> features:

Reviewer 2: I don't really understand vaults as well as you do so it might be helpful to ask Lena, but to me it seems like what we are looking for is a table of what **can't** be stored as a secret in the konnect config store, and two bullets that say licenses and the kong.conf values. Then you can do something like: 

* kong.conf values: You can’t reference secrets stored in a [Konnect Config Store](https://deploy-preview-385--kongdeveloper.netlify.app/how-to/configure-the-konnect-config-store/) Vault in kong.conf because Konnect resolves the secret after Kong Gateway connects to the control plane.

Because otherwise it looks like it supports everything but that.

Author: Good call out. I think we were also lacking more details about what fields were allowed, so I'll make it a list instead of a feature table and provide a better note about Config Store limitations.

Reviewer: List works fine too. Those are all the allowed values 😅 the DP certificates should never have come up - that was an error - but the rest is the accurate list.

  - title: |

>       All `kong.conf` values<sup>1</sup>
> 
>       For example:
>       * Data store usernames and passwords, used with PostgreSQL and Redis
>       * Private X.509 certificates

Reviewer 2: ```suggestion
      `kong.conf` values
```

    gcp: true
    aws: true
    azure: true
  - title: Certificates and keys stored in the Certificate Gateway entity
    url: /gateway/entities/certificate/
    konnect: true
    env: true
    hcv: true
    gcp: true
    aws: true
    azure: true
  - title: {{site.base_gateway}} license<sup>2</sup>
    url: /gateway/entities/license/
    konnect: false
    env: true
    hcv: true
    gcp: true
    aws: true
    azure: true
  - title: Referenceable plugin fields, such as third-party API keys (see table below for all values)
    konnect: true
    env: true
    hcv: true
    gcp: true
    aws: true
    azure: true
{% endfeature_table %}

{:.info}
> **Notes:**
> * <sup>1</sup>: You can't reference secrets stored in a [{{site.konnect_short_name}} Config Store](/how-to/configure-the-konnect-config-store/) Vault in `kong.conf` because {{site.konnect_short_name}} resolves the secret after {{site.base_gateway}} connects to the control plane.
> * <sup>2</sup>: In Konnect, the {{site.base_gateway}} license is managed and stored by {{site.konnect_short_name}}, and doesn't need to be stored manually in any Vault.

The following plugin fields can be stored and referenced as secrets:

> 

Reviewer 2: ```suggestion
## Plugins
```
So it stands out more. I'm also open to the idea of having an H2 plugins, that says what plugins can be used, and then links out to a doc that contains a full list of referenceable fields.

When you want to use a secret stored in a Vault, you can reference the secret with a `vault` reference. You can use the `vault` reference in places such as `kong.conf`, declarative configuration files, logs, or in the UI.

{:.info}

> > **Note:** You can't reference secrets stored in a [{{site.konnect_short_name}} Config Store](/how-to/configure-the-konnect-config-store/) Vault in `kong.conf` because {{site.konnect_short_name}} resolves the secret before connecting to {{site.base_gateway}}. 

Reviewer 3: ```suggestion
> **Note:** You can't reference secrets stored in a [{{site.konnect_short_name}} Config Store](/how-to/configure-the-konnect-config-store/) Vault in `kong.conf` because {{site.konnect_short_name}} resolves the secret after {{site.base_gateway}} connects to the control plane.
```

Reviewer 3: Gateway connects to Konnect, not the other way around

Author: I made these edits in my latest commits
